src/Microsoft.Diagnostics.DebugServices.Implementation/ThreadService.cs

```diff
@@ -18,6 +18,7 @@ namespace Microsoft.Diagnostics.DebugServices.Implementation
     public abstract class ThreadService : IThreadService
     {
         protected readonly ITarget Target;
+        private readonly IDisposable _onFlushEvent;
```

Comment:
It was easy to do the once in TargetFromDataReader.cs but disposing this and any event in other per-target services is more difficult and not necessary because once a target instance is closed there will be no more flush events.  All the per target services (most of them), would all be "collected" together once the root target instanced is unreferenced.

Reply:
Then we don't need to cache the registration. You just don't need them if WeakReferences go away.